lib/page-object/section.rb

```diff
@@ -0,0 +1,40 @@
+module PageObject
+  class Section < Elements::Element
+    extend Accessors
+    attr_reader :platform
+    #Given that page sections implement their own behaviors,
+    #we don't want misses to fall through to the watir element
+    undef_method :method_missing
+
+    private
+    #there is currently no better way to handle the dual nature of sections being like elements and pages
+    def call_block(&block)
+      block.arity == 1 ? block.call(self) : self.instance_eval(&block)
+    end
+  end
+  class SectionCollection
+    include Enumerable
+    attr_reader :sections
```

Comment:
I wasn't sure whether the sections of the collection should be openly available... I left it that way for now, but I'm not attached either way as of right now.